internal/app/singularity/plugin_compile_linux.go

```diff
@@ -53,20 +69,38 @@ func CompilePlugin(sourceDir, destSif string) error {
 }
 
 // buildPlugin takes sourceDir which is the string path the host which contains the source code of
-// the plugin. The output path is where the plugin .so file should
-// end up.
+// the plugin. buildPlugin returns the path to the built file, along with an error
 //
 // This function essentially runs the `go build -buildmode=plugin [...]` command
-func buildPlugin(sourceDir string) error {
-	baseDir := filepath.Base(sourceDir)
-	scmd := exec.Command("singularity", "run", "--cleanenv", "-B",
-		sourceDir+":"+filepath.Join(containedSourceDir, baseDir),
-		containerURI, baseDir)
-
-	scmd.Stderr = os.Stderr
-	scmd.Stdout = os.Stdout
-	scmd.Stdin = os.Stdin
-	return scmd.Run()
+func buildPlugin(sourceDir string) (string, error) {
+	// assuming that sourceDir is within trimpath for now
+	out := pluginObjPath(sourceDir)
+	c := fmt.Sprintf(buildtmpl, trimpath, out, sourceDir)
+	buildcmd := exec.Command("go", strings.Split(c, " ")...)
+
+	buildcmd.Dir = workpath
+	buildcmd.Stderr = os.Stderr
+	buildcmd.Stdout = os.Stdout
+	buildcmd.Stdin = os.Stdin
+
+	return out, buildcmd.Run()
+}
+
+// generateManifest takes the path to the plugin source, sourceDir, and generates
+// its corresponding manifest file.
+func generateManifest(sourceDir string) (string, error) {
+	in := pluginObjPath(sourceDir)
+	out := pluginManifestPath(sourceDir)
+
+	c := fmt.Sprintf(manifesttmpl, trimpath, mangenpath, in+" "+out)
```

Comment:
why not pass in+" "+out as two separate arguments?

Reply:
I have no reasonable answer to this question